pavelib/paver_tests/test_assets.py

```diff
@@ -202,3 +203,90 @@ def test_watch_theme_assets(self):
+      * expected_log_location: the expected string to be used for piping collectstatic logs
+    """
+
+    @ddt.data(
+        [{"expected_log_location": "> /dev/null"}],  # pipe to /dev/null by default
+        [{"debug": True, "expected_log_location": ""}],  # pipe to console in debug mode
+        [{"debug": False, "expected_log_location": "> /dev/null"}],
+        [{
+            "specified_log_location": "/foo/bar.log",
+            "expected_log_location": "> /foo/bar.log"
+        }],  # can use specified log location
+        [{"systems": ["lms", "cms"], "expected_log_location": "> /dev/null"}],  # multiple systems can be called
+    )
+    @ddt.unpack
+    def test_collect_assets(self, options):
+        """
+        Ensure commands sent to the environment for collect_assets are as expected
+        """
+        log_loc = options.get("expected_log_location", None)
+        systems = options.get("systems", ["lms"])
+        expected_messages = self._set_expected_messages(log_location=log_loc, systems=systems)
+        collect_assets(
+            systems,
+            "devstack",
+            debug=options.get("debug", None),
+            collectstatic_log=options.get("specified_log_location", None)
+        )
+        self.assertEqual(self.task_messages, expected_messages)
+
+    def _set_expected_messages(self, log_location, systems):
+        """
+        Returns a list of messages that are expected to be sent from paver
+         to the commandline for collectstatic functions. This list is constructed
+         based on the log location and systems being passed in.
+        """
+
+        expected_messages = []
+        for sys in systems:
+            expected_messages.append(
+                'python manage.py {system} --settings=devstack collectstatic --noinput {log_loc}'.format(
+                    system=sys,
+                    log_loc=log_location
+                )
+            )
+        return expected_messages
+
+
+@ddt.ddt
+class TestUpdateAssetsTask(PaverTestCase):
+    """
+    These are nearly end-to-end tests, because they observe output from the commandline request,
+    but do not actually execute the commandline on the terminal/process
+    """
+
+    @ddt.data(
+        [{"expected_substring": "> /dev/null"}],  # go to /dev/null by default
+        [{"cmd_args": ["--debug"], "expected_substring": "collectstatic --noinput "}]  # TODO: make this regex
+    )
+    @ddt.unpack
+    def test_update_assets_task_collectstatic_log_arg(self, options):
+        """
+        Scoped test that only looks at what is passed to the collecstatic options
+        """
+        cmd_args = options.get("cmd_args", [""])
+        expected_substring = options.get("expected_substring", None)
+        call_task('pavelib.assets.update_assets', args=cmd_args)
+        self.assertTrue(self._assert_substring_in_list(self.task_messages, expected_substring))
+
+    def _assert_substring_in_list(self, messages_list, expected_substring):
```

Comment:
@cpennington btw I couldn't find anything like that in the default unittest assertion library, unless you happen to know something that already does this?